File: MoveZeroes/MoveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        length = len(nums)
        zeros = 0
        
        i = 0
        while i < length:
            if nums[i] == 0:
                del nums[i]
                zeros += 1
                length -= 1
            else:
                i += 1
            
        for i in range(zeros):
            nums.append(0)

File: code/MoveZeroes/MoveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zeros = 0;
        
        for(vector<int>::iterator it = nums.begin(); it != nums.end(); it++){
            if(*it == 0){
                nums.erase(it);
                zeros++;
                it--;
            }
        }
        
        for(int i = 0; i < zeros; i++){
            nums.push_back(0);
        }
    }
};
